src/control/jobs/XournalScheduler.cpp

```diff
@@ -12,51 +12,60 @@ XournalScheduler::XournalScheduler()
 XournalScheduler::~XournalScheduler() = default;
 
 void XournalScheduler::removeSidebar(SidebarPreviewBaseEntry* preview) {
-    // Dont await for tasks, as it completly blocks whole program.
```

Comment:
I haven't noticed this blocking after making this change (possibly due to the refactor?), but I don't know how often/what caused this blocking to occur...

Reply:
Maybe, because u started to use "lock guards" which will automatically unlock the mutex when the guard is destroyed.